src/data/reports/data-quality/DataQualityDefaultRepository.ts

```diff
@@ -0,0 +1,88 @@
+import _ from "lodash";
+import { PaginatedObjects } from "../../../domain/common/entities/PaginatedObjects";
+import { DataQualityConfig, DataQualityItem } from "../../../domain/reports/data-quality/entities/DataQualityItem";
+import {
+    DataQualityOptions,
+    DataQualityRepository,
+} from "../../../domain/reports/data-quality/repositories/DataQualityRepository";
+import { D2Api, Pager } from "../../../types/d2-api";
+import { DataStoreStorageClient } from "../../common/clients/storage/DataStoreStorageClient";
+import { StorageClient } from "../../common/clients/storage/StorageClient";
+import { Instance } from "../../common/entities/Instance";
+
+export class DataQualityDefaultRepository implements DataQualityRepository {
+    private storageClient: StorageClient;
+    private globalStorageClient: StorageClient;
+
+    constructor(private api: D2Api) {
+        const instance = new Instance({ url: this.api.baseUrl });
+        this.storageClient = new DataStoreStorageClient("user", instance);
+        this.globalStorageClient = new DataStoreStorageClient("global", instance);
+    }
+
+    async getIndicators(options: DataQualityOptions, namespace: string): Promise<PaginatedObjects<DataQualityItem>> {
+        const dataQuality = await this.globalStorageClient.getObject<DataQualityConfig>(namespace);
+        const { paging, sorting } = options;
+
+        const dataQualityIndicatorErrors =
+            dataQuality?.validationResults?.filter(
+                r => (!r.denominatorresult || !r.numeratorresult) && r.metadataType === "Indicator"
+            ) ?? [];
+
+        const dataQualityIndicatorErrorsInPage = _(dataQualityIndicatorErrors)
+            .orderBy([row => row[sorting.field]], [sorting.direction])
+            .drop((paging.page - 1) * paging.pageSize)
+            .take(paging.pageSize)
+            .value();
+
+        const pager: Pager = {
+            page: paging.page,
+            pageSize: paging.pageSize,
+            pageCount: Math.ceil(dataQualityIndicatorErrors.length / paging.pageSize),
+            total: dataQualityIndicatorErrors.length,
+        };
+
+        return { pager: pager, objects: dataQualityIndicatorErrorsInPage };
+    }
+
+    async saveDataQuality(namespace: string, dataQuality: DataQualityItem[]): Promise<void> {
+        return await this.globalStorageClient.saveObject<DataQualityItem[]>(namespace, dataQuality);
+    }
```

Comment:
As mentioned, looks like this is unused.
Maybe i lack some context but, in case of using it, maybe its better to store `dataQualityIndicatorErrors` instead of `dataQuality` if the idea is to use the globalStorageClient as a kind of cache.